redundant-bmc/src/role_determination.hpp

```diff
@@ -18,16 +18,42 @@ namespace role_determination
 struct Input
 {
     size_t bmcPosition;
+    size_t siblingPosition;
+    Role siblingRole;
+    bool siblingHeartbeat;
+    bool siblingProvisioned;
+};
+
+/**
+ * @brief Role determination error cases.
+ */
+enum class ErrorCase
+{
+    noError,
+    internalError,
+    samePositions
+};
+
+/**
+ * @brief The role and the error reason returned from run()
+ */
+struct RoleInfo
+{
+    Role role;
+    ErrorCase error;
```

Comment:
Do you think adding an error message will be helpful when creating the PEL in the future? Or would capturing the last N journal traces from the RBMC and CFAM manager be more useful for error logging?

(I noticed you have some good traces from the role determination function, and since this is related to error handling, we could consider it in the future to assist with debugging).

Reply:
I have an upcoming commit to do that, it just didn't get pushed up in this PR.  It'll look like:

```
  if (!result && !input.siblingProvisioned)
    {
        result = {Role::Active, ErrorCase::noError,
                  "Sibling is not provisioned"};
    }

```

and then it will be saved in the journal and also in a persistent data file for debug.  I also have an upcoming 'rbmctool' which will print it.

Reply:
those debug data is not planned to capture in the PEL if we are planning to create for some error case? Just in journal and a separate persistent file which will be collected as part of the BMC dump?

Reply:
I think I have notes that we also have to create a PEL if we can't enable redundancy, so I may as well put the reasons in there too.